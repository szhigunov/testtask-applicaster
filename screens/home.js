import React, {useState, useEffect, useMemo} from 'react';
import {View, FlatList} from 'react-native';
import {FeedItem} from '../components/feed-item';
import {Preloader} from '../components/preloader';
import {NoResults} from '../components/no-results';
import {SearchInput} from '../components/search-input';

export const HomeScreen = ({navigation: {navigate}}) => {
  const [data, setData] = useState([]);
  const [searchInput, setSearchInput] = useState('');

  const filteredData = useMemo(() => {
    if (searchInput.length === 0) {
      return data;
    } else {
      let newData = data.filter(({title}) => title.includes(searchInput));
      return newData;
    }
  }, [data, searchInput]);

  useEffect(() => {
    Promise.all([
      fetch(
        'http://assets-production.applicaster.com/applicaster-employees/israel_team/Elad/assignment/link_json.json',
      ).then(res => res.json()),
      fetch(
        'http://assets-production.applicaster.com/applicaster-employees/israel_team/Elad/assignment/video_json.json',
      ).then(res => res.json()),
    ]).then(([links, videos]) => setData([...links.entry, ...videos.entry]));
  }, []);

  return data.length === 0 ? (
    <Preloader title="Loading..." />
  ) : (
    // eslint-disable-next-line react-native/no-inline-styles
    <View style={{flex: 1}}>
      <SearchInput onChangeText={setSearchInput} value={searchInput} />
      {filteredData.length === 0 && <NoResults />}
      <FlatList
        data={filteredData}
        renderItem={({item}) => <FeedItem onPress={navigate} item={item} />}
        keyExtractor={({id}) => id}
      />
    </View>
  );
};
