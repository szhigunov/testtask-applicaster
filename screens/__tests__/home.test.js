/**
 * @format
 */

import 'react-native';
import React from 'react';
import {HomeScreen} from '../home';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('HomeScreen renders correctly', () => {
  renderer.create(<HomeScreen navigation={{navigate: jest.fn()}} />);
});
