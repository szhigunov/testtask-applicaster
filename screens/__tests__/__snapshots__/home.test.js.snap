// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
    <Text
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      Loading...
    </Text>
  </View>
</View>
`;
